components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/service/IdentityKeyStoreGeneratorImpl.java

```diff
@@ -0,0 +1,334 @@
+            } else {
+
+                String msg = "Error when adding a keyStore";
+                LOG.error(msg, e);
+                throw new KeyStoreManagementException(msg, e);
+            }
+        } catch (Exception e) {
+
+            String msg = "Error when processing keystore/pub. cert to be stored in registry";
+            LOG.error(msg, e);
+            throw new KeyStoreManagementException(msg, e);
+        }
+    }
+
+    /**
+     * This method generates the keypair and stores it in the keystore
+     *
+     * @param keyStore A keystore instance
+     * @throws KeyStoreManagementException Error when generating key pair
+     */
+    private void generateContextKeyPair(KeyStore keyStore, String context)
+            throws KeyStoreManagementException {
+        try {
+            CryptoUtil.getDefaultCryptoUtil();
+            //generate key pair
+            String keyGenerationAlgorithm = getKeyGenerationAlgorithm();
+            KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance(keyGenerationAlgorithm);
+            int keySize = getKeySize(keyGenerationAlgorithm);
+            if (keySize != 0) {
+                keyPairGenerator.initialize(keySize);
+            }
+            KeyPair keyPair = keyPairGenerator.generateKeyPair();
+
+            // Common Name and alias for the generated certificate
+            String commonName = "CN=" + tenantDomain +
+                    KEY_STORE_CONTEXT_SEPARATOR + context + ", OU=None, O=None, L=None, C=None";
+
+            //generate certificates
+            X500Name distinguishedName = new X500Name(commonName);
+
+            Date notBefore = new Date(System.currentTimeMillis() - 1000L * 60 * 60 * 24 * 30);
+            Date notAfter = new Date(System.currentTimeMillis() + (1000L * 60 * 60 * 24 * 365 * 10));
+
+            SubjectPublicKeyInfo subPubKeyInfo = SubjectPublicKeyInfo.getInstance(keyPair.getPublic().getEncoded());
+            BigInteger serialNumber = BigInteger.valueOf(new SecureRandom().nextInt());
+
+            X509v3CertificateBuilder certificateBuilder = new X509v3CertificateBuilder(
+                    distinguishedName,
+                    serialNumber,
+                    notBefore,
+                    notAfter,
+                    distinguishedName,
+                    subPubKeyInfo
+            );
+
+            String algorithmName = getSignatureAlgorithm();
+            JcaContentSignerBuilder signerBuilder =
+                    new JcaContentSignerBuilder(algorithmName).setProvider(getJCEProvider());
+            PrivateKey privateKey = keyPair.getPrivate();
+            X509Certificate x509Cert = new JcaX509CertificateConverter().setProvider(getJCEProvider())
+                    .getCertificate(certificateBuilder.build(signerBuilder.build(privateKey)));
+
+            //add private key to KS
+            keyStore.setKeyEntry(tenantDomain +KEY_STORE_CONTEXT_SEPARATOR + context,
```

Comment:
```suggestion
            keyStore.setKeyEntry(tenantDomain + KEY_STORE_CONTEXT_SEPARATOR + context,
```

Reply:
addressed in https://github.com/wso2/carbon-identity-framework/pull/6189/commits/6ca0b3d01b63c8f3f8ee122988af923b204f342b